src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -477,6 +473,8 @@ private function validateAttempts(callable $interviewer, OutputInterface $output
                 throw $e;
             } catch (\Exception $error) {
             }
+
+            $attempts = $attempts ?? -(int) $this->isTty();
```

Comment:
Was the motivation to move this inside a loop because isTty() result may change between iterations? I wonder how can that happen

Reply:
`echo "$several_answers" | the-command` breaks in real life, with `stream_isatty()` emitting PHP warnings `"data lost during stream conversion"`.

Moving after reading the response means we never hit this line when correct responses are provided. Which is the case in the described situation (programmatic filling of answers).